splitn/main.py

```diff
@@ -0,0 +1,85 @@
+from itertools import chain, combinations, repeat
+from collections.abc import Generator
+from copy import deepcopy
+from random import randint
+from re import compile
+
+import typer
+
+class CombinationsGenerator():
+    def __init__(self, seed: str, separator: str) -> None:
+        self.seed = seed
+        self.separator = separator
+        self.length = len(self.seed)
+        self.digits = [digit for digit in self.seed]
+
+    def combinations(self) -> Generator[str, None, None]:
+        for k in range(0, self.length):
+            if not k:
+                yield self.seed
+            else:
+                for pattern in combinations(range(1, self.length), k):
+                    yield self.split_n(pattern)
+
+    def split_n(self, pattern: tuple[int, ...]) -> str:
+        combination = deepcopy(self.digits)
+        for position, i in zip(pattern, range(len(pattern))):
+            combination.insert(position + i, self.separator)
+        return "".join(combination)
+
+class RandomNumbersGenerator():
+    def __init__(self, length:int) -> None:
+        self.length = length
+
+    def random_number(self) -> str:
+        return "".join([str(randint(0,9)) for _ in repeat(None, self.length)])
```

Comment:
Did you think about using just a plain function instead of a class with a single method?